portal/src/main/webapp/js/src/oncoprint/webgl/src/js/oncoprintmodel.js

```diff
@@ -144,6 +144,11 @@ var OncoprintModel = (function () {
 	this.track_active_rules = {}; // from track id to active rule map (map with rule ids as keys)
 	this.track_info = {};
 	this.track_has_column_spacing = {}; // track id -> boolean
+	this.track_expansion_init_callback = {}; // track id -> function that initializes expansion data if applicable
+	this.track_expansion_genes = {}; // track id -> array of gene data to expand if initialized
+	this.track_expansion_index = {}; // track id -> index in track_expansion_genes[track_id] if initialized
```

Comment:
I don't understand the need for this field `track_expansion_index` - it seems extremely hacky to me.